internal/core/permission.go

```diff
@@ -11,27 +12,28 @@ import (
 
 // PermissionService impelements the permission.Permission interface.
 type PermissionService struct {
-	dataprovider dataprovider.DataProvider
+	externalDataprovider dataprovider.ExternalDataProvider
 }
 
 // NewPermissionService returns a new permission service.
 func NewPermissionService(externalDataprovider dataprovider.ExternalDataProvider) *PermissionService {
-	dp := dataprovider.NewDataProvider(externalDataprovider)
-	return &PermissionService{dp}
+	return &PermissionService{externalDataprovider}
```

Comment:
I would give the function attribute `externalDataprovider` a shorter name. The scope of the variable is only two lines. So a name like `edp` is not a problem.

See also: https://github.com/golang/go/wiki/CodeReviewComments#variable-names